app/controllers/auth_controller.rb

```diff
@@ -20,8 +20,16 @@ def create
     when :google
       user = GoogleAuthService.find_user_by_token(data_attrs[:google_id_token], data_attrs[:create_user])
       AuthToken.new(user: user)
-    when :okta, :facebook
-      user = "::#{method.to_s.capitalize}AuthService".constantize.find_user_by_token(data_attrs[:"#{method}_access_token"], data_attrs[:create_user])
+    when :okta
+      user = OktaAuthService.find_user_by_token(data_attrs[:okta_access_token], data_attrs[:create_user])
+      AuthToken.new(user: user)
+    when :facebook
+      # Handle both access tokens and ID tokens for Facebook
+      if data_attrs[:facebook_access_token]
```

Comment:
[Suggested tests] Add test cases for edge scenarios: both facebook tokens provided, neither provided. Evidence: Current tests only cover single token scenarios. Impact: Missing validation coverage. [Category: File]